### ISSUE TITLE
ODBC.jl returns bad values for negative bigints with Vertica

### ISSUE THREAD

**Issue by dhazeghi** [JuliaDatabases/ODBC.jl#250]
So I'm running into an issue with ODBC.jl and the Vertica driver.  I do not see this issue using the same Vertica ODBC driver in python, so I believe the issue is in ODBC.jl.  Basically, any negative values in a bigint column are not fetched - I get random garbage in their place

```
zdb=> select bigint01 from d.t_test;
 bigint01 
----------
        1
         
       -1
       -3
        2
(5 rows)
```

```
julia> columnValues    = ODBC.query(dsn,"SELECT bigint01 FROM d.t_test")
5×1 DataFrames.DataFrame
│ Row │ bigint01        │
│     │ Int64⍰          │
├─────┼─────────────────┤
│ 1   │ 1               │
│ 2   │ missing         │
│ 3   │ 140401082606736 │
│ 4   │ 140401649603280 │
│ 5   │ 2               │
```

When I enabled ODBC tracing, the log shows the following:
```
[ODBC][15289][1570587711.948849][SQLSetStmtAttrW.c][554]
                Exit:[SQL_SUCCESS]
[ODBC][15289][1570587711.948870][SQLFetchScroll.c][159]
                Entry:
                        Statement = 0x2f30680
                        Fetch Orentation = 1
                        Fetch Offset = 0
[ODBC][15289][1570587711.949068][SQLFetchScroll.c][353]
                Exit:[SQL_SUCCESS_WITH_INFO]
                DIAG [22003] [Vertica][Support] (40470) Conversion error at column 1 and row 3: Numeric value out of range.

                DIAG [22003] [Vertica][Support] (40470) Conversion error at column 1 and row 4: Numeric value out of range.
```

**Comment by dhazeghi** [comment]
After initially thinking the Vertica ODBC driver was at fault, I wrote a test program in C to exercise the same APIs.  It worked.  But, I realized that it was using different datatypes in the SQLBindCols call than ODBC.jl.  After I adjusted those, the C program failed too.

The issue is the mapping of SQL_BIGINT to SQL_C_UBIGINT (-27).   @quinnj  - I see this code has been there ever since the beginning.  Is there a reason why BIGINT should be assumed to be unsigned (unlike other types which are signed)?  Thanks.
